File: csharp/DiagramPrinter/PrintQueue.cs
namespace DiagramPrinter;

public class PrintQueue
{
    private readonly List<PrintMetadata> _items;
    private readonly PhysicalPrinter _printer;

    public PrintQueue(PhysicalPrinter physicalPrinter)
    {
        this._printer = physicalPrinter;
        this._items = new List<PrintMetadata>();
    }

    public virtual void Add(PrintMetadata data)
    {
        this._items.Add(data);
    }
}

File: java/src/main/java/sammancoaching/PrintQueue.java
package sammancoaching;

import java.util.ArrayList;
import java.util.List;

public class PrintQueue {
    private List<PrintMetadata> _items = new ArrayList<PrintMetadata>();
    private PhysicalPrinter _printer;

    public PrintQueue(PhysicalPrinter physicalPrinter) {
        this._printer = physicalPrinter;
    }

    public void add(PrintMetadata data) {
        this._items.add(data);
    }
}
